### DISCUSSION TITLE
Join Transform Question

### DISCUSSION THREAD

**Discussion by psheehymn** [grafana/grafana-infinity-datasource#637] [Q&A] [answered]
We are using an Infinity data source to interface with the Adobe Analytics API. I have several queries that all return different counts related to the same set of error messages. Unfortunately, the numbers must each come back in a separate query, but the list of error messages is the same for each query result..

I want to join these in a single table, joining on the error message. In the Columns section, I ensured that the error message fields are all named identically and the counts all have different names: 
![SC1](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/93273674/cfee30ef-ba05-454d-8510-cf859696b295)

However, in the selector for the join transform, it prepends the field names with the query data frame name, resulting in the names NOT being identical and thus unavailable for the join:

![SC2](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/93273674/f450748a-f2aa-447d-89f0-abe46b9554a4)

Is this a behavior of the the Infinity plugin or something native to Grafana? If the former, is there any way to circumvent this behavior?

Thanks!

**Comment by yesoreyeram** [comment, marked as answer]
What you needed was "Merge" transformation rather instead "Join by field"

example : 

<img width="1060" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/153843/b9dad739-91ac-47d9-9d25-f687c8f329a6">
<img width="1281" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/153843/69d176e2-9cc0-475f-a14f-db20d418c09f">

**Comment by psheehymn** [comment]
Merge worked. Thank you!

We are in the process of upgrading our Grafana instance and plan to move to move to the newer Infinity plugin at that time.

Thanks again!
